### DISCUSSION TITLE
server multi client

### DISCUSSION THREAD

**Discussion by GiulioZ94** [KoljaB/RealtimeSTT#32] [Q&A]
Is there a way to use one istance of AudioToTextRecorder in a websocket server to handle multiple client at same time?

**Comment by KoljaB** [comment]
Currently there is no option to parallelize the execution of transcription or voice activity detection. I don't think the underlying libraries (esp faster-whisper) support this, so I think you'd have to write your own wrapper around RealtimeSTT prob using multiprocessing.

**Reply by GiulioZ94** [reply]
I believe it would be fantastic to have a faster whisper server, along with integrating Realtime STT capabilities with the server. Similarly, implementing Realtime TTS using a Coqui TTS server and integrating it.

**Reply by KoljaB** [reply]
Managing the load for such multiple client scenarios will be challenging. You can assign different GPUs to [faster-whisper](https://github.com/SYSTRAN/faster-whisper/blob/1eb9a8004c509a4af2960955374520c35b7b793a/faster_whisper/transcribe.py#L107) to get some degree of parallelism. Stays questionable tho because you'll end up in a 'one GPU per user' ratio, at least for a worst case load scenario.

I'm only developing RealtimeSTT and RealtimeTTS as side projects for my Linguflex assistant btw, so that limits my focus a bit. Currently I don't plan any bigger developments around the infrastructure libs.
